iac-extensions/spring-config/src/main/java/org/sonar/iac/springconfig/parser/yaml/SpringConfigYamlPreprocessor.java

```diff
@@ -27,23 +28,23 @@ public class SpringConfigYamlPreprocessor {
   private static final Pattern LINE_SEPARATORS = Pattern.compile("(\n|\r|\r\n|\u2028|\u2029)");
   private static final String DOCUMENT_BREAK = "---";
   private static final int DOCUMENT_BREAK_LENGTH = DOCUMENT_BREAK.length();
+  private static final Pattern MAVEN_VARIABLE_AT_TOKEN_START = Pattern.compile("([^:]++:\\p{javaWhitespace}*)@(?<name>[^@]++)@(.*+)");
+  private static final Predicate<String> containsInlineCommentAfterDocumentBreak = line -> line.startsWith(DOCUMENT_BREAK);
+  private static final Predicate<String> containsMavenVariableAtTokenStart = line -> MAVEN_VARIABLE_AT_TOKEN_START.matcher(line).matches();
 
   public String preprocess(String source) {
     var lines = LINE_SEPARATORS.split(source, -1);
 
-    if (Stream.of(lines).noneMatch(SpringConfigYamlPreprocessor::containsInlineCommentAfterDocumentBreak)) {
+    if (Stream.of(lines).noneMatch(containsInlineCommentAfterDocumentBreak.or(containsMavenVariableAtTokenStart))) {
       return source;
     }
 
     return Stream.of(lines)
       .map(SpringConfigYamlPreprocessor::removeInlineCommentAfterDocumentBreak)
+      .map(SpringConfigYamlPreprocessor::transformMavenSubstitutions)
       .collect(Collectors.joining("\n"));
```

Comment:
We lose the original line separator.
This should not be an issue, although I still want to raise awareness about it (this was an issue for docker, where the preprocessor was changing the offset, changing a `\r\n` into `\n` would have higher impact).